### PR TITLE
Remove native types from generated C++ code

### LINKED ISSUE #391: Replace uses of platform-specific types in the generated C++ code
For example, port indices should use `FwIndexType` instead of `NATIVE_INT_TYPE`. This is a first step towards eliminating unnecessary uses of platform-specific types in the F Prime code base. That in turn will make the framework code better (and it will eliminate a lot of static analysis warning noise in pull requests to nasa/fprime).

### PR DESCRIPTION
**Pull request by bocchino**
Closes #391.

### CODE REVIEW

**Review by LeStarch — APPROVED** [review]
@bocchino  there were two types I struggled to find in the examples.  Otherwise I approve!

File: compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentEvents.scala [hunk 1]
```diff
@@ -64,7 +64,7 @@ case class ComponentEvents (
         linesClassMember(
           Line.blank :: lines(
             s"""|//! Throttle for ${event.getName}
-                |NATIVE_UINT_TYPE ${eventThrottleCounterName(event.getName)};
+                |FwIndexType ${eventThrottleCounterName(event.getName)};
```
[the remark this answers is not in the document]
Comment (bocchino) [reply compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentEvents.scala:67]: This is an event throttle counter, i.e., the number of times an event has been emitted. It is bounded by the throttle value, which feels like an index.

File: compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentInputPorts.scala [hunk 2]
```diff
@@ -130,7 +130,7 @@ case class ComponentInputPorts(
           lines(
             s"""|// Serialize message ID
                 |_status = $bufferName.serialize(
-                |  static_cast<NATIVE_INT_TYPE>(${portCppConstantName(p)})
+                |  static_cast<FwEnumStoreType>(${portCppConstantName(p)})
```
[the remark this answers is not in the document]
Comment (bocchino) [reply compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentInputPorts.scala:133]: This is serializing a message ID, which is a C++ enum value, so we use FwEnumStoreType.

[the remark this answers is not in the document]
Comment (bocchino) [reply compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentInputPorts.scala:133]: Example:
```
[I] bocchino@MT-306179 ~/J/T/f/c/t/f/t/c/base (issue-391-native-types)> grep -n 'Serialize message ID' * | head -n 1
ActiveAsyncProductPortsOnlyComponentAc.cpp:329:  // Serialize message ID
```

**Comment by LeStarch** [comment]
Approve!

**Pull request merged**